internal/bminventory/inventory.go

```diff
@@ -1224,7 +1240,17 @@ func (b *bareMetalInventory) InstallClusterInternal(ctx context.Context, params
 		log.Infof("generated ignition for cluster %s", cluster.ID.String())
 
 		log.Infof("Storing OpenShift cluster ID of cluster %s to DB", cluster.ID.String())
-		err = b.storeOpenshiftClusterID(ctx, cluster.ID.String())
+		var updatedCluster *common.Cluster
+		if updatedCluster, err = b.storeOpenshiftClusterID(ctx, *cluster.ID); err != nil {
```

Comment:
I think we could do this without the extra cluster object. There's no need to do the extra db call to fetch the updated cluster if we already have the information we need for the next call. Would it be better to extract the "get openshift cluster id" logic out of `storeOpenshiftClusterID`, call that first then call these next two methods with that id?

Really we don't need a cluster object, we just need the ams id, openshift cluster id, and our cluster id, right?

Reply:
I started with it and changed it back for 2 reasons:
1. we also need `cluster.amsSubscriptionId` and I preferred fetching it from DB rather than using the cluster that is captured by the closure - this is not an "eager" fetch anyway.
2. I wanted to stay consistent with all other `integrateWithAms*` method that gets a `common.Cluster`

If you still think it is better to change it, I will get the `AmsSubscriptionID` from the current cluster object and the `openshiftClsuterID` as a return value of `storeOpenshiftClusterID`

Reply:
Yeah, that would work too. I think extracting like I described would be the cleanest option though.
We should avoid unnecessary calls to the DB whenever possible so either way would be better than what we have now in my opinion.

Reply:
Updated.